compiler/rustc_llvm/llvm-wrapper/RustWrapper.cpp

```diff
@@ -111,9 +112,24 @@ extern "C" void LLVMRustSetNormalizedTarget(LLVMModuleRef M,
   unwrap(M)->setTargetTriple(Triple::normalize(Triple));
 }
 
-extern "C" void LLVMRustPrintPassTimings() {
-  raw_fd_ostream OS(2, false); // stderr.
-  TimerGroup::printAll(OS);
+extern "C" const char *LLVMRustPrintPassTimings(void) {
+  std::string buf;
+  raw_string_ostream SS(buf);
+  TimerGroup::printAll(SS);
+  SS.flush();
+  char* CStr = (char*) malloc((buf.length() + 1) * sizeof(char));
+  strcpy(CStr, buf.c_str());
+  return CStr;
+}
```

Comment:
I would do something like this:
```suggestion
extern "C" const char *LLVMRustPrintPassTimings(size_t *Len) {
  std::string buf;
  raw_string_ostream SS(buf);
  TimerGroup::printAll(SS);
  SS.flush();
  *Len = bug.length();
  char *CStr = malloc(*Len);
  memcpy(CStr, buf.c_str(), *Len);
  return CStr;
}
```
And then use from_raw_parts on the Rust side to create he `[u8]` slice.